packages/jest-leak-detector/src/index.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+'use strict';
+
+const v8 = require('v8');
+const vm = require('vm');
+const weak = require('weak');
+
+// Setting the flag has no effect on the main context, but will allow new
+// contexts to access the garbage collector.
+v8.setFlagsFromString('--expose-gc');
+const gc = vm.runInNewContext('gc');
+
+const PRIMITIVE_TYPES = new Set([
+  'undefined',
+  'boolean',
+  'number',
+  'string',
+  'symbol',
+]);
+
+export default class {
+  _held: boolean;
+
+  constructor(value: ?Object) {
+    if (this._isPrimitive(value)) {
+      throw new TypeError('Primitives cannot leak memory');
+    }
+
+    weak(value, () => (this._held = false));
+    this._held = true;
+
+    // Ensure value is not leaked by the closure created by the "weak" callback.
+    value = null;
+  }
+
+  isLeaked(): boolean {
```

Comment:
`isLeaked` doesn't make so much sense. I propose `hasLeak`, `hasLeaked`, `didDetectLeak`, `contextHasLeak` or any variation of those.

Reply:
`has` has a meaning about something that happened in the past; and its result will not vary; which is not true. Something that is leaking right now might not leak just immediately after cleaning it; and the result will change. For instance:

```javascript
const foo = {};
const detector = new LeakDetector(foo);

console.log(detector.hasLeaked()); // true
foo = null;
console.log(detector.hasLeaked()); // false
```

To me, naming the method `hasLeaked` would make me think that the second call would still return `true` (it leaked once, so it will always leaked).  On the other side, `isLeaked` talks about the status at that precise moment in time. That said, I'm not a native English speaker, so I might be wrong 😅

Reply:
You are right about checking the current state, but then `isLeaked` is still wrong as it would be `isLeaking`.

Reply:
`isLost`? :)

Reply:
I think "leak" is a clearer term. I'll change to `isLeaking` :)